### PR TITLE
github actions: use GitHub Actions for checking build

### PR DESCRIPTION
**Pull request by kenhys**
ChangeLog: Because of new pricing model,
https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing
It may be better to use another CI/CD such as GitHub Actions.

### CODE REVIEW

**Comment by mrunge** [comment]
Thank you, I saw this, and it seems it is still blocked in a travis-ci job.I don't want to steal your PRs, for testing purposes, I've opened https://github.com/collectd/collectd/pull/3786, (including both of your commits), but it is still blocked by travis.

**Comment by kenhys** [comment]
To disable Travis-CI, we need to remove .travis-ci.yml.

But This PR is aimed to gradually migrate to GitHub Actions
because it does not fully support the same functionality yet.

**Comment by kenhys** [comment]
I've fixed a syntax error and pushed again.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            libudev-dev \
+            libupsclient-dev \
+            libvarnish-dev \
+            libvirt-dev \
+            libxen-dev \
+            libxml2-dev \
+            libyajl-dev \
+            linux-libc-dev \
+            openjdk-11-jdk-headless \
+            perl
```
Comment (mrunge) [inline .github/workflows/linux.yml:77]: missing \ ?

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            protobuf-c-compiler
```
Comment (mrunge) [inline .github/workflows/linux.yml:78]: same here?

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    name: Build Check
+    strategy:
+      fail-fast: false
+      matrix:
+        label:
+          - "GCC"
+          - "Clang"
+      include:
+        - label: "Clang"
+          CC=clang
```
Comment (mrunge) [inline .github/workflows/linux.yml:16]: In my tests, CC=lang is not valid yaml.

**Review by mrunge — COMMENTED** [review]
I left a few comments, I wonder how this passed.

**Comment by kenhys** [comment]
Sorry, I've missed that already Cirrus CI is adopted so it may be ok to use it.

**Pull request closed without merging**